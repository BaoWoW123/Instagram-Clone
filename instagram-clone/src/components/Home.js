import "../styles/Home.css";
import { testImgArr, testPostArr } from "./testImages";
import post from "../assets/navBar/post.png";
import bookmark from "../assets/navBar/bookmark.png";
import send from "../assets/navBar/send.png";
import postmessage from "../assets/navBar/postmessage.png";
import like from "../assets/navBar/heart.png";

const Home = () => {
    
  const showNext = () => {
    const followers = document.querySelectorAll(".follower");
    followers.forEach((el) => {
      return (el.style.transform = "translateX(-360px)");
    });
    const prevBtn = document.querySelector(".storiesPrevBtn");
    const nextBtn = document.querySelector(".storiesNextBtn");
    nextBtn.className = "storiesNextBtn hidden";
    prevBtn.className = "storiesPrevBtn";
  };

  const showPrev = () => {
    const followers = document.querySelectorAll(".follower");
    followers.forEach((el) => {
      return (el.style.transform = "translateX(0px)");
    });
    const prevBtn = document.querySelector(".storiesPrevBtn");
    const nextBtn = document.querySelector(".storiesNextBtn");
    nextBtn.className = "storiesNextBtn";
    prevBtn.className = "storiesPrevBtn hidden";
  };

  return (
    <div className="Home" aria-label="Home">
      <div className="feed">
        <div className="storiesContainer">
          <button className="storiesNextBtn" onClick={showNext}>
            {">"}
          </button>
          <button className="storiesPrevBtn hidden" onClick={showPrev}>
            {"< "}
          </button>
          <div className="storiesWrapper">
            {testImgArr.map((el, i) => {
              return (
                <div className="follower" data-id={i}>
                  <div className="followerImg">
                    <img src={el.img} />
                  </div>
                  <div>{el.username}</div>
                </div>
              );
            })}
          </div>
        </div>
        <div className="posts">
          {testPostArr.map((el, i) => {
            return (
              <div className="post">
                <div>
                  <div className="postInfo">
                    <img src={el.img} />
                    <div style={{ fontWeight: "bold" }}>{el.username}</div>
                  </div>
                  <img src={post} />
                </div>
                <img src={el.post} />
                <div className="postDescrip">
                  <div className="postNavBar">
                    <div>
                      <img src={like} />
                      <img src={postmessage} />
                      <img src={send} />
                    </div>
                    <div>Nav Dots here</div>
                    <div>
                      <img src={bookmark} />
                    </div>
                  </div>
                  <div className="postLikes">
                    <div>Images</div>
                    <div>
                      Like by <b>username</b> and <b>others</b>
                    </div>
                  </div>
                  <div className="postCaption">
                    <div>
                      <b>{el.username}</b> Some captions here and there or maybe
                      everywhere
                    </div>
                    <div>View {Math.ceil(Math.random() * 10)} comments</div>
                  </div>
                  <div className="postComment">
                    <input type="text" placeholder="Add a comment..."></input>
                    <button>Post</button>
                  </div>
                </div>
              </div>
            );
          })}
        </div>
      </div>
      <div className="rightDiv">
        <div>
          <div className="suggestSwitchAcc">
            <img src={testImgArr[0].img} />
            <div>
              <div>Account Username</div>
              <div>Some username</div>
            </div>
          </div>
          <button>Switch</button>
        </div>
        <div className="suggestions">
          <div>
            Suggestions For You{" "}
            <button style={{ color: "black" }}>See All</button>
          </div>
          {testImgArr.map((el, i) => {
            if (i < 5) {
              return (
                <div className="suggestionFollowers">
                  <div>
                    <img src={el.img} />
                    <div>
                      <div className="suggestFollowerName">{el.username}</div>
                      <div>
                        Followed by {el.username} + {i + 3} more
                      </div>
                    </div>
                  </div>
                  <button>Follow</button>
                </div>
              );
            }
          })}
        </div>
      </div>
    </div>
  );
};

export default Home;
