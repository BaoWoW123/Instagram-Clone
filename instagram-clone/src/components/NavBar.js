import { Link } from "react-router-dom"
import home from "../assets/navBar/home.png";
import search from "../assets/navBar/search.png";
import explore from "../assets/navBar/explore.png";
import message from "../assets/navBar/message.png";
import notification from "../assets/navBar/notification.png";
import create from "../assets/navBar/create.png";
import more from "../assets/navBar/more.png";
import reels from "../assets/navBar/reels.png";
import { testImgArr } from "./testImages";
import { useEffect } from "react";

const NavBar = (props) => {
    const showMore = () => {
        const moreMenu = document.querySelector(".moreMenu");
        if (moreMenu.className === "moreMenu hidden") {
          moreMenu.className = "moreMenu";
          moreMenu.style.backgroundColor = "lightgray";
        } else {
          moreMenu.className = "moreMenu hidden";
          moreMenu.style.backgroundColor = "none";
        }
      };

    return (
        <div className="header">
        <div className="title">Itstagram</div>
        <div className="navBar">
          <div>
            <Link to="/home">
              <img src={home} />
              Home
            </Link>
          </div>
          <div>
            <Link to="/home">
              <img src={search} />
              Search
            </Link>
          </div>
          <div>
            <Link to="/home">
              {" "}
              <img src={explore} />
              Explore
            </Link>
          </div>
          <div>
            <Link to="/home">
              {" "}
              <img src={reels} />
              Reels
            </Link>
          </div>
          <div>
            <Link to="/home">
              <img src={message} />
              Messages
            </Link>
          </div>
          <div>
            <Link to="/home">
              <img src={notification} />
              Notifications
            </Link>
          </div>
          <div>
            <Link to="/post">
              <img src={create} />
              Create
            </Link>
          </div>
          <div>
            <Link to="/profile">  
                <img src={(props.userInfo) ? props.userInfo.profileImg : testImgArr[3].img} style={{width:'48px', height:'48px', borderRadius:'50%'}}/>
                Profile</Link>
          </div>
          <div className="moreWrapper">
            <div>
              <div className="moreMenu hidden">
                <div>Settings</div>
                <div>Your Activity</div>
                <div>Saved</div>
                <div>Switch Appearance</div>
                <div>Report a problem</div>
                <div>Switch Accounts</div>
                <div onClick={props.signOut}>Log out</div>
              </div>
              <div onClick={showMore}>
                <img src={more} />
                More
              </div>
            </div>
          </div>
        </div>
      </div>
    )
}

export default NavBar