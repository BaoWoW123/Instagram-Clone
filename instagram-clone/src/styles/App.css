body,
html {
  margin: 0;
}

.App {
  display: flex;
  height: 100%; /* temporary */
  font-size: 17px;
  font-family: sans-serif;
}

.App > :last-child {
  flex-grow: 1;
  background-color: whitesmoke;
}

.header {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  border-right: 2px solid lightgray;
  gap: 20px;
  position: sticky;
  top: 0;
  height: 100vh;
}

.title {
  padding-top: 50px;
  padding-left: 20px;
}

.navBar {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
}

.navBar > * {
  width: 220px;
  padding-top: 16px;
}

.navBar> div {
  padding-right: 100px;
}

.navBar > * > * {
  display: flex;
  align-items: center;
}
.navBar > * > a, .moreWrapper>div>:last-child{
  padding-left: 20px;
  gap: 5px;
}

.moreWrapper {
  margin-bottom: 0;
  margin-top: auto;
}

.moreWrapper > div {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.moreWrapper> div > :last-child{
  display: flex;
  align-items: center;
}

.moreMenu {
  display: flex;
  flex-direction: column;
  width: 100%;
  border-radius: 3px;
}
.moreMenu > * {
  padding: 11px 0px;
  padding-left: 20px;
}
.moreWrapper > * >*:hover {
  cursor: pointer;
}

a {
  text-decoration: none;
  color: inherit;
}

.hidden {
  display: none;
}
