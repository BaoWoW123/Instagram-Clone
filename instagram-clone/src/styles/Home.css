.Home {
  display: grid;
  grid-template-columns: 1fr 5fr 4fr;
}

.feed {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  gap: 1.5rem;
}

/* FOLLOWER'S STORIES */

.storiesContainer {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  height: 20vh;
  position: relative;
}

.storiesWrapper {
  overflow: hidden;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  border: 2px solid lightgray;
  background-color: white;
  border-radius: 10px;
  width: 36rem;
  gap: 10px;
  padding: 15px 0px;
  position: relative;
}

.storiesWrapper > :first-child {
  padding-left: 10px;
}

.storiesContainer > button {
  background-color: beige;
  border: beige;
  border-radius: 50%;
  font-weight: bold;
  padding: 5px 10px;
}

.storiesPrevBtn {
  position: absolute;
  left: 10px;
  z-index: 1;
}

.storiesNextBtn {
  position: absolute;
  right: 10px;
  width: min-content;
  z-index: 1;
}

.follower {
  display: flex;
  flex-direction: column;
  text-align: center;
  font-size: 20px;
  max-width: 80px;
  transition: 0.2s;
}

.followerImg {
  display: flex;
  align-content: center;
  justify-content: center;
  background: linear-gradient(45deg, yellow, red, purple);
  border-radius: 100%;
  padding: 4px;
}
.followerImg > img {
  width: 60px;
  height: 60px;
  border-radius: 100%;
  padding: 3px;
  background-color: white;
}

/* FEED */
.posts > * {
  display: flex;
  justify-content: center;
  align-items: center;
  border: 2px solid lightgray;
  border-radius: 5px;
  width: 36rem;
  background-color: white;
}
.post {
  margin-bottom: 20px;
  display: flex;
  flex-direction: column;
}

.postInfo {
  display: flex;
  align-items: center;
}

.postInfo > :last-child {
  display: flex;
  align-items: center;
  padding: 0px 10px;
}

.post > :first-child {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 95%;
  padding: 10px 0px;
}
.post > :first-child > :last-child {
  display: flex;
  align-content: center;
  justify-content: center;
}

.post > :first-child > :first-child > img {
  height: 50px;
  border-radius: 50%;
}

.post > img {
  width: 100%;
  height: 100%;
}

.followersNextBtn {
  position: absolute;
}

.postDescrip {
  display: flex;
  flex-direction: column;
  width: 100%;
  position: relative;
}

.postNavBar {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  align-items: center;
  text-align: center;
  width: 100%;
  padding: 10px 0px;
}

.postNavBar > :first-child {
  display: flex;
  padding-left: 10px;
  gap: 10px;
}

.postNavBar > :last-child {
  display: flex;
  justify-content: flex-end;
  padding-right: 10px;
}

.postLikes {
  display: flex;
  gap: 10px;
  padding-left: 15px;
}

.postCaption {
  padding: 10px 15px;
}

.postComment {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-top: 1px solid rgb(0, 0, 0, 0.5);
}

.postComment > input {
  border: none;
  font-size: 20px;
  background-color: transparent;
  font-weight: bold;
}
.postComment > input::placeholder {
  color: rgb(0, 0, 0, 0.4);
}
input:focus {
  outline: none;
}
.postComment > * {
  padding: 20px;
}

.postComment > *:hover {
  opacity: 1;
}

.postComment button {
  background-color: transparent;
  border: none;
  font-size: 20px;
  color: dodgerblue;
  font-weight: bold;
  opacity: 0.4;
}
/* RIGHT DIV */
.rightDiv {
  display: flex;
  flex-direction: column;
  margin-top: 2rem;
  margin-left: 2rem;
  width: 25rem;
}

.rightDiv > * button {
  align-items: flex-end;
  border: none;
  background-color: transparent;
  color: dodgerblue;
  font-size: large;
  font-weight: 600;
}

.rightDiv > :first-child > * img {
  width: 70px;
  border-radius: 50%;
}

.suggestSwitchAcc {
  display: flex;
  flex: 1;
  gap: 15px;
}
.suggestSwitchAcc > :last-child > :first-child {
  font-weight: bold;
}

.suggestSwitchAcc > :last-child > :last-child {
  color: gray;
  font-weight: bold;
}
.suggestSwitchAcc > :last-child {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

.suggestions > * {
  padding: 10px 0px;
}

.suggestionFollowers > :first-child {
  display: flex;
  gap: 15px;
  flex: 1;
}

.suggestionFollowers > * img {
  border-radius: 50%;
  width: 50px;
  height: 50px;
}

.suggestions > *,
.rightDiv > :first-child {
  display: flex;
  justify-content: space-between;
}

.suggestionFollowers > :first-child > :last-child {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}
.suggestFollowerName {
  font-weight: bold;
}
.suggestFollowerName + div {
  color: gray;
  font-size: 15px;
  font-weight: bold;
}
