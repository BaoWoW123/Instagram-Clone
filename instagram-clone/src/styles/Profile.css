.profilePage {
    display: grid;
    grid-template-columns: 1fr 8fr;
}

.profile {
    display: flex;
    flex-direction: column;
    padding-top: 50px;
}
.topProfile {
    display: grid;
    grid-template-columns: 1fr 2fr;
    margin: 0px 100px;
    border-bottom: 2px solid lightgray;
}
.profileInfo {
    display: flex;
    flex-direction: column;
    gap: 10px;
}
.profileInfo>:first-child {
    display: flex;
    justify-content: flex-start;
    gap: 20px;
    font-size: 30px;
}
.profileInfoBtns {
    display: flex;
    gap: 10px;
    flex-wrap: wrap;
}

.profileInfoBtns > *:not(:last-child) {
    border: none;
    background-color: dodgerblue;
    color: white;
    font-size: large;
    padding: 10px 20px;
    border-radius: 5px;
}

.profileInfoBtns > :last-child {
    border: none;
    background-color: transparent;
    font-weight:bold ;
    font-size: 30px;
}

.profileInfoBtns > button:hover {
    cursor: pointer;
}

.topProfile img {
    margin: auto;
    padding-bottom: 40px;
    width: 200px;
    height: 200px;
    border-radius: 50%;
}

.topProfile>div {
    font-size: 20px;
}

.userPostsNavBar {
    grid-column: 1/4;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 100px;
    top: 0;
    padding: 20px 0px;
    width: 100%;
    font-size: 20px;
}

.userPostsNavBar>* {
    opacity: .6;
}

.userPostsNavBar>*:hover {
    opacity: 1;
    cursor: pointer !important;
}

.userPosts {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-gap: 40px;
    padding: 0px 100px;
    position: relative;
}

.userPost {
    position: relative;
    color: white;
    cursor: pointer;
}

.userPost:hover .userPostInfo {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    position: absolute;
    top: 0;
    width: 100%;
    height: 100%;
    background-color: rgb(0, 0, 0, 0.6);
}
.userPost > img {
    width: 100%;
    height: 100%;
    object-fit: cover;
}
.userPostInfo {
    display: none;
}

.userPostInfo div {
    display: inline-block;
    font-weight: bold;
    font-size: 1.7rem;
}
