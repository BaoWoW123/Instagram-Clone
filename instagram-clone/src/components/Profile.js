import "../styles/Profile.css";
import NavBar from "./NavBar";
import likes from "../assets/navBar/likes.png";
import comment from "../assets/navBar/comment.png";
import { testImgArr } from "./testImages";
import Cropper from "react-easy-crop";
import { useEffect, useState } from "react";
import { async } from "@firebase/util";
import { doc, setDoc } from "firebase/firestore";
import { auth, database } from "../firebase";

const Profile = (props) => {
  let [crop, setCrop] = useState({ x: 0, y: 0 });
  let [zoom, setZoom] = useState(1);
  let [profileImgCrop, setProfileImgCrop] = useState(null);
  let [profileImgPixels, setProfileImgPixels] = useState({});

  const showProfilePopup = (e) => {
    console.log("popup");
    const profilePopupWrapper = document.querySelector(".profilePopupWrapper");
    const profilePopupOptions = document.querySelector('.profilePopupOptions')
    if (profilePopupWrapper.className === "profilePopupWrapper") {
      profilePopupWrapper.className = "profilePopupWrapper hidden";
    } else {
      profilePopupWrapper.className = "profilePopupWrapper";
      profilePopupOptions.className = 'profilePopupOptions'
    }
  };

  const selectProfileImg = (e) => {
    console.log("change");
    const imgInput = document.querySelector(".profileImgUpload");

    imgInput.click();
  };

  const updateProfileImg =  async () => {
    const profilePopupOptions = document.querySelector(".profilePopupOptions");
    const profileImgUpload = document.querySelector('.profileImgUpload')
    const cropper = document.querySelector('.profileImgCropperWrapper')
    cropper.className = 'profileImgCropperWrapper'
    profilePopupOptions.className = "profilePopupOptions hidden";
    console.log("img selected");
    if (profileImgUpload.files.length) {
      await setProfileImgCrop(URL.createObjectURL(profileImgUpload.files[0]))
    }

  };

   const createProfileImg = async () => {
    let img = new Image();
    const canvas = document.querySelector(".profileCanvas");
    const ctx = canvas.getContext("2d");
    const profilePopupWrapper = document.querySelector('.profilePopupWrapper')
    const profileImgCropperWrapper = document.querySelector('.profileImgCropperWrapper') 

    img.src = profileImgCrop;
    profilePopupWrapper.className = "profilePopupWrapper hidden";
    profileImgCropperWrapper.className = 'profileImgCropperWrapper hidden'; 


    await ctx.drawImage(
      img,
      profileImgPixels.x, profileImgPixels.y,
      profileImgPixels.width, profileImgPixels.height,
      0,0,
      300,
      150
    );
    
    const canvasURL = canvas.toDataURL('image/png');
    setProfileImgCrop(canvasURL)
    await setDoc(
      doc(database, "users", props.user.uid), {profileImg: canvasURL}
    );
    props.getUserInfo(auth.currentUser)
  };
  const onCropComplete = (croppedArea, cropperAreaPx) => {
    setProfileImgPixels(cropperAreaPx);
  };

  useEffect(() => {
    const unsubscribe = auth.onAuthStateChanged((user) => {
      props.setUser(user);
    });

    return unsubscribe();
  }, [])

  return (
    <div className="profilePage" aria-label="Profile">
      <NavBar signOut={props.signOut} userInfo = {props.userInfo}/>
      <div className="profile">
        <div className="profilePopupWrapper hidden">
          <span className="profilePopup">
            <div className="profilePopupOptions">
            <div>
              <h2>Change profile image</h2>
            </div>
            <div onClick={selectProfileImg}>Upload image</div>
            <div onClick={showProfilePopup}>Cancel</div>
            </div>
            <input
              type="file"
              className="profileImgUpload"
              onChange={updateProfileImg}
              style={{ display: "none" }}
              accept="image/*"
            />        
          </span>
          <div className="profileImgCropperWrapper hidden">
              <div>
              <Cropper
                image={profileImgCrop}
                crop={crop}
                zoom={zoom}
                aspect={1}
                onCropChange={setCrop}
                onCropComplete={onCropComplete}
                onZoomChange={setZoom}
                cropShape='round'
              />
              </div>
              <button type="button" onClick={createProfileImg}>Save</button>
              <canvas className="profileCanvas hidden"></canvas>
            </div>
        </div>
        <div className="topProfile">
          <img
            src={props.userInfo ? props.userInfo.profileImg : testImgArr[3].img}
            onClick={showProfilePopup}
            style={{position:'top'}}
          />
          <div className="profileInfo">
            <div>
              {props.user.displayName}
              <div className="profileInfoBtns">
                {" "}
                {/* SHOULDN'T DISPLAY ON OWN PROFILE */}
                <button type="button">Follow</button>
                <button type="button">{"v"}</button>
                <button type="button">···</button>
              </div>
            </div>
            <div style={{ display: "flex", padding: "20px 0px", gap: "50px" }}>
              <div>100 posts</div>
              <div>300 followers</div>
              <div>500 following</div>
            </div>
            <div
              style={{ display: "flex", flexDirection: "column", gap: "5px" }}
            >
              <div>{props.user.displayName}</div>
              <div>Insert bio here</div>
              <div>aWebsite.com</div>
            </div>
          </div>
        </div>
        <div>
          <div className="userPostsNavBar">
            <div>Posts</div>
            <div>Reels</div>
            <div>Videos</div>
            <div>Tagged</div>
          </div>
          <div className="userPosts">
            {testImgArr.map((el, i) => {
              return (
                <div className="userPost" key={i}>
                  <img src={el.img} />
                  <div className="userPostInfo">
                    <div style={{ display: "flex", alignItems: "center" }}>
                      <img src={likes} /> {i + 1}
                    </div>
                    <div style={{ display: "flex", alignItems: "center" }}>
                      <img src={comment} /> {i + 3}
                    </div>
                  </div>
                </div>
              );
            })}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Profile;
