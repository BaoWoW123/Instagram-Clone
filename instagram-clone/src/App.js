import React, { useEffect, useState } from "react";
import { Routes, Route, Link, useNavigate } from "react-router-dom";
import Home from "./components/Home";
import Login from "./components/Login";
import Post from "./components/Post";
import Profile from "./components/Profile";
import "./styles/App.css";
import Signup from "./components/Signup";
import {
  getAuth,
  onAuthStateChanged,
  GoogleAuthProvider,
  signInWithPopup,
  signOut,
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
} from "firebase/auth";
import { auth, db } from "./firebase";

const App = () => {

  let [user, setUser] = useState(); 
  const navigate = useNavigate();

  const logInEmail = (email, pw) => {
    signInWithEmailAndPassword(getAuth(), email, pw).then((userInfo) => {
      console.log('signed in', userInfo)
      navigate('/home')
    }).catch((error) => {
      alert(error)
    })
  }
  const createAcc = (email, pw) => {
    createUserWithEmailAndPassword(getAuth(), email, pw).then((userInfo) => {
      //signed in
      console.log('created email acc', userInfo)
      navigate('/home')
    }).catch((error) => {
      alert(error)
    })
  }

  const routeSignup = () => {
    navigate('/signup')
  }

  const routeLogin = () => {
    navigate('/login')
  }

  const signInGoogle = async () => {
    const provider = new GoogleAuthProvider();
    await signInWithPopup(auth, provider);
    setUser(user = auth)//user resets on restart
    navigate("/home")
    setTimeout(function(){updatePage(auth)},1)//delayed updatePage(), can't await navigate()
  };
  const signOutAcc = () => {
    const moreMenu = document.querySelector(".moreMenu");
    if (moreMenu.className === "moreMenu") {
      moreMenu.className = "moreMenu hidden";
    }
    try {
      signOut(getAuth());
      navigate("/login");
    } catch (error) {
      console.log("signout error", error);
    }
  };

  const updatePage = (auth) => {
    let acc = document.querySelector(".rightDivAcc");
    let accImg = document.querySelector('.accImg')
    accImg.src = auth.currentUser.photoURL;
    acc.firstChild.textContent = auth.currentUser.displayName;
    acc.lastChild.textContent = auth.currentUser.email;
  };

  const firebaseAuth = () => { 
    onAuthStateChanged(getAuth());
  };

  useEffect(() => {
    
  }, [])

  
  return (
    <div className="App">
      <Routes>
        <Route index path='/' element={<Login signInGoogle={signInGoogle} logInEmail={logInEmail} routeSignup={routeSignup}  />} />
        <Route path='/login' element={<Login signInGoogle={signInGoogle} logInEmail={logInEmail}  routeSignup={routeSignup}/>} />
        <Route path="/post" element={<Post signOut={signOutAcc}/>} />
        <Route path="/profile" element={<Profile signOut={signOutAcc}/>} />
        <Route path="/home" element={<Home signOut={signOutAcc}/>} />
        <Route path="/signup" element={<Signup createAcc={createAcc} routeLogin={routeLogin} signInGoogle={signInGoogle}/>} />
      </Routes>
    </div>
  );
};

export default App;
