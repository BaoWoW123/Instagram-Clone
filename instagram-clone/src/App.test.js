import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import App from "./App";
import Home from "./components/Home";
import { BrowserRouter, Routes } from "react-router-dom";
import NavBar from "./components/NavBar";
import Profile from "./components/Profile";

describe("App renders component after click event", () => {
  test("snapshot test", () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test("show home page", () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
    expect(screen.getByLabelText("Home").className).toBe("Home");
  });
  test("simulate login with fake account", async () => {
    const submit = jest.fn()
    render(
      <BrowserRouter>
      <App submit={submit}/>
      </BrowserRouter>
    );
    const emailInput = screen.getByPlaceholderText("Email");
    const pwInput = screen.getByPlaceholderText('Password')
    const loginBtn = screen.getByText('Log in')
    userEvent.type(emailInput, 'test@gmail.com');
    userEvent.type(pwInput, 'test123');
    userEvent.click(loginBtn)
    expect((await screen.findByLabelText('Home')).className).toBe('Home')
  });
});
