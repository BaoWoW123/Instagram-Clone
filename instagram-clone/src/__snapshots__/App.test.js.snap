// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component after click event snapshot test 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="loginPage"
    >
      <div>
        <fieldset>
          <form
            class="loginForm"
          >
            <h1>
              Itstagram
            </h1>
            <input
              class="email"
              placeholder="Email"
              type="email"
            />
            <div
              class="passwordDiv"
            >
              <input
                class="password"
                placeholder="Password"
                type="password"
              />
              <button
                type="button"
              >
                Show
              </button>
            </div>
            <button
              class="loginBtn"
            >
              Log in
            </button>
            <h2>
              <span
                class="OR"
              >
                OR
              </span>
            </h2>
            <span
              class="googleLogin"
            >
              <img
                src="googleLogo.png"
              />
              <div>
                Login with Google
              </div>
            </span>
            <div>
              Forgot password?
            </div>
          </form>
        </fieldset>
        <fieldset>
          <div
            class="signUp"
          >
            Don't have an account? 
            <button>
              Sign up
            </button>
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
